components/ingest-service/projectupdater/manager.go

```diff
@@ -25,88 +26,126 @@ const (
 	maxNumberOfConsecutiveFails          = 10
 )
 
+type stage struct {
+	state           string
+	projectUpdateID string
+	esJobID         string
+}
+
 // Manager - project update manager
 type Manager struct {
-	state                 string
-	projectUpdateID       string
-	esJobID               string
+	// stage should only be updated by using the 'updateStage' function
+	stage                 stage
 	percentageComplete    float32
 	estimatedEndTimeInSec int64
 	client                backend.Client
 	authzProjectsClient   iam_v2.ProjectsClient
 	eventServiceClient    automate_event.EventServiceClient
+	configManager         *config.Manager
+	updateQueue           chan<- func(stage) stage
 }
 
-// TODO
-// * Store running job IDs so if service restart it can pick up where it left off
-
 // NewManager - create a new project update manager
 func NewManager(client backend.Client, authzProjectsClient iam_v2.ProjectsClient,
-	eventServiceClient automate_event.EventServiceClient) Manager {
-	return Manager{
-		state:               notRunningState,
+	eventServiceClient automate_event.EventServiceClient, configManager *config.Manager) *Manager {
+
+	updateQueue := make(chan func(stage) stage, 100)
+	manager := &Manager{
+		stage: stage{
+			state: notRunningState,
+		},
 		client:              client,
 		authzProjectsClient: authzProjectsClient,
 		eventServiceClient:  eventServiceClient,
+		configManager:       configManager,
+		updateQueue:         updateQueue,
 	}
+
+	// Single goroutine that updates the stage data
+	go manager.stageUpdater(updateQueue)
+
+	manager.resumePreviousState()
```

Comment:
After the `stage` data is read from the config file restore any running processes.